### PR TITLE
Donzh add exception handling

### PR DESCRIPTION
**Pull request by donzh**

### CODE REVIEW

**Comment by donzh** [comment]
@reyang Could you please help me review this pull request when you are free? Thanks!

**Comment by donzh** [comment]
@yaohaizh Could you please help me review this pull request. reyang is busy? Thanks!

File: src/devdisco.js [hunk 1]
```diff
@@ -204,6 +204,17 @@ function cmd_list_usb_uart() {
   UsbUartTransport.beginDiscovery(onDeviceDiscovered);
 }
 
-bi.start();
-main(process.argv.slice(1));
-bi.flush();
+var d = require('domain').create();
```
Comment (reyang) [inline src/devdisco.js]: need to have a reasonable name rather than 'd'

File: src/lib/ethernet-transport.js [hunk 2]
```diff
@@ -11,6 +12,9 @@ function resolveIpAddressToHostNameAsync(address) {
   return new vow.Promise(function(resolve, reject /*, notify */) {
     dns.reverse(address, function(err, domains) {
       if(err) {
+        bi.trackEvent('ethernet_error', {
```
Comment (reyang) [inline src/lib/ethernet-transport.js:15]: please check with Kevin on how to distinguish error, warning and log

Comment (donzh) [reply src/lib/ethernet-transport.js:15]: At present the bi doesn't distinguish error, warning and log.

File: src/devdisco.js (hunk 1 above)
Comment (donzh) [reply src/devdisco.js]: Accept

File: src/lib/ethernet-transport.js (hunk 2 above)
Comment (donzh) [reply src/lib/ethernet-transport.js:15]: At present we use the key 'error', 'warning' and 'info' to represent error, warning and log.

File: src/devdisco.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ var VERSION = require('../package.json').version;
 
 function main(argv) {
   bi.trackEvent('command_line_arguments', {
-    argv: argv
+    info: argv.join(' ')
```
Comment (reyang) [inline src/devdisco.js:33]: should use JSON.stringify

**Review by reyang — APPROVED** [review]

**Pull request merged**